developing/dapps/get_started/cosmwasm-testnet.md

```diff
@@ -1,4 +1,321 @@
+cargo install cargo-run-script
+```
+
+### Setup Osmosis Testnet
+
+You can easily set up an Osmosis Testnet environment using the [Osmosis Installer](https://github.com/osmosis-labs/osmosis-installer). 
+
+Run the following and choose option #2 (Client Node) and #2 (Testnet) in order.
+
+```bash
+curl -sL https://get.osmosis.zone/install > i.py && python3 i.py
+```
+![](https://user-images.githubusercontent.com/70956926/172289722-0254d766-45d8-4491-b958-6e165464bb21.png)
+![](https://user-images.githubusercontent.com/70956926/172292272-3e75acc7-3f2e-4e5c-94ef-cb46401a4174.png)
+![](https://user-images.githubusercontent.com/70956926/172292397-130a1467-928a-45e9-ad6c-7ac0a1b6f420.png)
+Now you have successfully completed setting up an Osmosis client node in Testnet. In order to use `osmosisd` from the cli, either reload your terminal or refresh your profile with : `‘source ~/.profile’`
+
+### Setup the Client
+
+First, create a wallet with the command following:
+
+```bash
+# add wallets for testing
+osmosisd keys add wallet
+```
+
+When you run the commands above, `osmosisd` will prompt you all the information related to that wallet in YAML (.yml) format.
+
+```bash
+- name: wallet
+  type: local
+  address: osmo1v9yrqx8aaddlna29zxngr4ye3jnxtpprrej532
+  pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AmKFqbczx7j/sYlqO2irXUSsQMdEN9Ugg1W2AOm7knh3"}'
+  mnemonic: ""
+
+**Important** write this mnemonic phrase in a safe place.
+It is the only way to recover your account if you ever forget your password.
+
+divert cliff issue spirit penalty chief improve neck enjoy pipe sing loop inherit behind space next tourist acid axis easy never ball enemy moment
+```
+
+You need some tokens named `OSMO`(`uosmo`) in your address to interact with the network.
+
+You must participate in the [Osmosis discode](https://discord.com/invite/osmosis) to request a faucet of the Osmosis Testnet. After gaining access to the testnet channel on the `#roles` channel of the discode, you can request a testnet token by sending the following message on the `#faucet` channel:
+
+```bash
+$request <address>
+```
+![](https://user-images.githubusercontent.com/70956926/172293039-b832fd96-d62d-44de-9889-ee400a9ec815.png)
+![](https://user-images.githubusercontent.com/70956926/172293080-aee186ad-ef51-43bb-ac6e-1a8b65c9ce04.png)
+
+
+Then, you can check that your faucet request has been successful by checking the balance of your wallet bank account by trying the command:
+
+```bash
+osmosisd query bank balances $(osmosisd keys show -a wallet)
+```
+
+- `osmosisd query bank balances [address]` commands query the total balance of an account.
+- `osmosisd keys show -a wallet` commands returns the address of the wallet that you created.
+
+
+
+## Deploy a Smart Contract
+### Clone cw-template
+
+For this example, we will use the [**cw-template**](https://github.com/InterWasm/cw-template) (CosmWasm Template) repo with counter example.
```

Comment:
Osmosis actually has a cw_template https://github.com/osmosis-labs/cw-tpl-osmosis for you to use... the deploy process is also in the docs.